pulsar-functions/transforms/src/main/java/org/apache/pulsar/functions/transforms/TransformResult.java

```diff
@@ -0,0 +1,22 @@
+package org.apache.pulsar.functions.transforms;
+
+import lombok.Getter;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.pulsar.client.api.Schema;
+
+@Getter
+public class TransformResult {
+    private final Schema<?> schema;
+    private final GenericRecord avroValue;
```

Comment:
it is not clear to me about how we deal with the Key.
maybe it is only a matter of documenting.

the Key will be part of the KeyValue payload (and so it will be inside the Object), and this is why we do not have a "avroKey" here

but if we are chaining the transformations, the "next" transformation will need the Key.

I wonder if it will be clearer to have:
```
class TransformResult<KeyNativeType, ValueNativeType> {
      Schema<?> keySchema;
      KeyNativeType  keyNativeValue;
      
      Schema<?> valueSchema;
      ValueNativeType  valueNativeValue;
      
}
```

So your pipeline will look like:
- transform the raw Input of the function (Message, unpack KeyValue...) to a TransformResult
- TransformResult -> apply one step -> TransformResult. (loop for every step)
- TransformResult -> raw Pulsar message (pack a new KeyValue if needed)

Advantages of this approach:
- each Transform works always on a Pair (key, value)
- we can deal also with the natural String key in case of non KeyValue input Message
- we can make configurable the latest step and allow any possible output (KeyValue, INLINE/SEPARATED, AVRO, JSON...)

Reply:
> it is not clear to me about how we deal with the Key.
Currently there's no modification of the key. The key is passed from transform to transform in the `value` Object field and used to recompose the KeyValue at the end.

> but if we are chaining the transformations, the "next" transformation will need the Key.
We get the key  from the `value` Object field which is unchanged between transforms.

> transform the raw Input of the function (Message, unpack KeyValue...) to a TransformResult
I had done that originally but if the transformations result in no transformation, I thought that we could avoid unpacking/repacking. But maybe that's overdoing it...

Reply:
Also with this structure of TransformResult, we convert the Avro schema to NATIVE_AVRO even if there were no modifications done by the transformation steps. Idk if that's optimal or not.

Reply:
done.
There are now properties to know if a schema was modified.